src/mscorlib/src/System/String.Manipulation.cs

```diff
@@ -1001,9 +1001,8 @@ public String[] Split(char[] separator, int count, StringSplitOptions options)
         [System.Security.SecuritySafeCritical]
         private unsafe String[] SplitInternal(char separator, int count, StringSplitOptions options)
         {
-            char* pSeparators = stackalloc char[1];
-            pSeparators[0] = separator;
-            return SplitInternal(pSeparators, /*separatorsLength*/ 1, count, options);
+            const int separatorsLength = 1;
```

Comment:
Nit - I think that the original code - where 1 was passed in as direct argument - read better.